data-es6.js

```diff
@@ -3715,6 +3739,44 @@ exports.tests = [
       */},
       res: temp.destructuringResults,
     },
+    'object destructuring expression': {
+      exec: function() {/*
+        var a, b, obj = { a:1, b:2 };
+        return ({a,b} = obj) === obj;
+      */},
+      res: {
+        tr:           true,
+        _6to5:        true,
+        firefox16:    true,
+      },
+    },
+    'chained object destructuring': {
+      exec: function() {/*
+        var a,b,c,d;
+        ({a,b}) = {c,d} = {a:1,b:2,c:3,d:4};
+        return a === 1 && b === 2 && c === 3 && d === 4;
+      */},
+      res: {
+        firefox16:    true,
+        es6tr:        true,
+      },
+    },
+    'throws on null and undefined': {
+      exec: function(){/*
+        var n = null;
```

Comment:
Actually, nevermind - it turns out the compile error was for some other aspect of the test's structure (namely, that `var {a}` appeared twice, which for some reason it didn't like) which has since been rewritten.